internal/lock.go

```diff
@@ -149,11 +160,131 @@ func (l *Lock) Download(dir string, tags []string, notags []string, perm string)
 		return fmt.Errorf("nothing to download")
 	}
 	errorCh := make(chan error, total)
-	for _, r := range filteredResources {
+
+	byteCh := make(chan int64)
+	resourceSizes := make([]int64, 0, len(filteredResources))
+
+	if bar {
+		startTime := time.Now()
```

Comment:
This whole logic belongs in its own file. I suggest refactoring this as a struct so that it is well encapsulated, easier to test and to use.